com.unity.ads.ios-support/Runtime/Plugins/iOS/SkAdNetworkManager.m

```diff
@@ -32,23 +32,43 @@ + (SkAdNetworkManager *)sharedInstance {
 }
 
 - (void)updateConversionValue:(NSInteger)conversionValue {
-    if (!self.isAvailable)
+    if (!self.isAvailable) {
+        NSLog(@"updateConversionValue::Not properly initialized");
         return;
+    }
 
-    SEL requestSelector = NSSelectorFromString(@"updateConversionValue:");
-    if ([self.SkAdNetworkClass respondsToSelector:requestSelector]) {
-        NSNumber *val = [NSNumber numberWithInteger:conversionValue];
-        [self.SkAdNetworkClass performSelector:requestSelector withObject:val];
+    id target = self.SkAdNetworkClass;
+    SEL selector = NSSelectorFromString(@"updateConversionValue:");
+    NSMethodSignature * signature = [target methodSignatureForSelector:selector];
+    if (signature) {
+        NSLog(@"updateConversionValue::Invoking conversionValue=%ld", conversionValue);
+        NSInvocation * invocation = [NSInvocation invocationWithMethodSignature:signature];
+        invocation.target = target;
+        invocation.selector = selector;
+        [invocation setArgument:&conversionValue atIndex:2];
+        [invocation invoke];
+    } else {
+        NSLog(@"updateConversionValue::Unknown method signature");
     }
```

Comment:
The only 2 cents I would add here is to reuse the code:

```objc
- (void)callFuntionReflectively: (NSString*)methodName args: NSArray * args {
   id target = self.SkAdNetworkClass;
  SEL selector = NSSelectorFromString(methodName);
  NSMethodSignature * signature = [target methodSignatureForSelector:selector];
  if (!signature) {
    return ;
  }
     NSInvocation * invocation = [NSInvocation invocationWithMethodSignature:signature];
      invocation.target = target;
      invocation.selector = selector;
     for (int i = 2, i < args.count + 2, i++) {
        [invocation setArgument:arg[I] atIndex:i];
      }
    [invocation setArgument:&conversionValue atIndex:2];
}
```


and then call it like: 
```objc
- (void)updateConversionValue:(NSInteger)conversionValue {
  [self callFuntionReflectively: @"updateConversionValue:" args: @[@(conversionValue)]];
}

- (void)registerAppForAdNetworkAttribution {
 [self callFuntionReflectively: @"registerAppForAdNetworkAttribution:" args: @[]];
}
```

Reply:
I was wondering that myself, but while this more verbose, I thought this was quite readable. I would prefer not to touch it.

This is mostly because a) I have no experience in Objective-C and reflection calls do not give you immediate feedback, and b) this is quite a pain to test e2e as you need a proper ad signature and attribution flow. I would hate to break code that currently works (for instance, I know it was written quickly, but your piece of code does not actually call invoke) and if we do not do the full e2e we could very well deliver without noticing it (iirc the previous version had a type in the function name and it was delivered anyways).

If you have a good test plan to propose, I will happily work with you on that. Otherwise the current code does the trick.

Reply:
>If you have a good test plan to propose, I will happily work with you on that. Otherwise the current code does the trick.

Means you already tested it, right? I guess the suggested approach can be tested the same way.

Reply:
As stated in the description, we released it as part of the GG package and went the full length to test it in a real game before contributing it back. I don't expect we will release a new version and follow-up with clients for a cosmetic change. That said, we will gladly go back to using this package in a future release if the issue is fixed.